python/ray/_private/services.py

```diff
@@ -384,49 +403,57 @@ def remaining_processes_alive():
     return ray.worker._global_node.remaining_processes_alive()
 
 
-def validate_redis_address(address):
-    """Validates address parameter.
+def canonicalize_bootstrap_address(addr: str):
+    """Canonicalizes Ray cluster bootstrap address to ip:port.
+    Reads address from the environment if needed.
+
+    This function should be used to process user supplied Ray cluster address,
+    via ray.init() or `--address` flags, before using the address to connect.
 
     Returns:
-        redis_address: string containing the full <host:port> address.
-        redis_ip: string representing the host portion of the address.
-        redis_port: integer representing the port portion of the address.
+        Ray cluster address string in <host:port> format.
     """
+    if not addr or addr == "auto":
+        addr = get_ray_address_from_environment()
+    try:
+        bootstrap_address = address_to_ip(addr)
+    except Exception:
+        logger.exception(f"Failed to convert {addr} to host:port")
+        raise
+    return bootstrap_address
 
-    if address == "auto":
-        address = find_redis_address_or_die()
-    redis_address = address_to_ip(address)
 
-    redis_address_parts = redis_address.split(":")
-    if len(redis_address_parts) != 2:
-        raise ValueError("Malformed address. Expected '<host>:<port>'.")
-    redis_ip = redis_address_parts[0]
+def extract_ip_port(bootstrap_address: str):
+    address_parts = bootstrap_address.split(":")
+    if len(address_parts) != 2:
+        raise ValueError(f"Malformed address {bootstrap_address}. "
+                         "Expected '<host>:<port>'.")
+    ip = address_parts[0]
     try:
-        redis_port = int(redis_address_parts[1])
+        port = int(address_parts[1])
     except ValueError:
-        raise ValueError("Malformed address port. Must be an integer.")
-    if redis_port < 1024 or redis_port > 65535:
-        raise ValueError("Invalid address port. Must "
-                         "be between 1024 and 65535.")
+        raise ValueError(f"Malformed address port {port}. Must be an integer.")
+    if port < 1024 or port > 65535:
+        raise ValueError(f"Invalid address port {port}. Must be between 1024 "
+                         "and 65535 (inclusive).")
+    return ip, port
 
-    return redis_address, redis_ip, redis_port
 
-
-def address_to_ip(address):
+def address_to_ip(addr: str):
     """Convert a hostname to a numerical IP addresses in an address.
 
     This should be a no-op if address already contains an actual numerical IP
     address.
 
     Args:
-        address: This can be either a string containing a hostname (or an IP
+        addr: This can be either a string containing a hostname (or an IP
```

Comment:
better to change it to address

Reply:
There is a function in this file also named as `address()`. My linter warns about variable name shadowing. I will revert this later.